### ISSUE TITLE
Failed to download sphinx.windows-x86_64.exe using sphinx-maven-plugin v2.x

### ISSUE THREAD

**Issue by donatelloOo** [trustin/sphinx-maven-plugin#9]
It seems the v2.x plug-in  tries to download a "sphinx.windows-x86_64.exe" file, which cannot be downloaded through maven.

I can reproduce the issue with each 2.x version (with slightly different error messages):

``` console
[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.0:generate (rtd) on project documentation: Failed to run the report: failed to download Sphinx binary at: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: 'posix:permissions' not supported as initial attribute -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.1:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.2:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.1.0:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.2.0:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.2.0/sphinx.windows-x86_64.exe: Connection timed out: connect -> [Help 1]
```

Could it be due to HTTPS proxy restrictions/misconfiguration on Maven or JVM ?
Note that I can download this file using my browser... but I cannot using a simple wget on same network:

``` console
$ wget https://github.com/trustin/sphinx-binary/releases/download/v0.2.0/sphinx.windows-x86_64.exe
--2018-04-24 16:27:59--  https://github.com/trustin/sphinx-binary/releases/download/v0.2.0/sphinx.windows-x86_64.exe
Resolving github.com (github.com)... 192.30.253.113, 192.30.253.112
Connecting to github.com (github.com)|192.30.253.113|:443... failed: Connection timed out.
Connecting to github.com (github.com)|192.30.253.112|:443... connected.
OpenSSL: error:14077419:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert access denied
Unable to establish SSL connection.
```
Could you provide a simple HTTP URL instead, to avoid this kind of SSL issues ?

**Comment by donatelloOo** [comment]
In fact I can reproduce the issue with each 2.x version (with slightly different error messages)

``` console

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.0:generate (rtd) on project documentation: Failed to run the report: failed to download Sphinx binary at: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: 'posix:permissions' not supported as initial attribute -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.1:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.0.2:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.1.0:generate (rtd) on project documentation: Failed to run the report: failed to download: https://github.com/trustin/sphinx-binary/releases/download/v0.1.1/sphinx.windows-x86_64.exe: Received fatal alert: access_denied -> [Help 1]

[ERROR] Failed to execute goal kr.motd.maven:sphinx-maven-plugin:2.2.0:generate (rtd) on project docwnload/v0.2.0/sphinx.windows-x86_64.exe: Connection timed out: connect -> [Help 1]

Could it be due to entreprise proxy/security restrictions ?

**Comment by trustin** [comment]
I believe it's an issue with your network provider or policy. You may want to host the files somewhere accessible from your network and specify the 'binaryUrl' option.

**Comment by trustin** [comment]
Ah, you said you can download successfully from your browser. Perhaps your browser has some additional root certificates installed or your JDK is using outdated root certificates?

**Comment by trustin** [comment]
Anyways, I do see some differences between downloading from a web browser (redirects to S3) and from other tools (no redirection), which may explain why. Let me dig a little bit more later. Of you have some spare time, it would be great if you do some research on what part of an HTTP request triggers a redirection to S3 and let me know that.

**Comment by donatelloOo** [comment]
I'm trying to run SphinxRunnerTest from the source code.
I will try to give you the detailed issue.

**Comment by trustin** [comment]
Much appreciated. Let me stay tuned to your findings.

**Comment by donatelloOo** [comment]
It seems that it's definitely a proxy issue, which for an unknown reason cannot reuse the system proxies, even if we set `-Djava.net.useSystemProxies=true`.

Using manual settings will do the job:
``` java
System.setProperty("https.proxyHost", "web-proxy.my.company.net");
System.setProperty("https.proxyPort", "8080");
System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
```
But I cannot find a way to provide JVM parameters to plugin context during the maven build.
Could you help ?

**Comment by donatelloOo** [comment]
From another approach, why the sphinx binaries are not deployed on a maven repository, in a more standard way, so that we could reuse maven settings.xml / default proxy configuration ?

**Comment by trustin** [comment]
That's really interesting. I just use `URLConnection` which should honer JVM settings. Does this work for you: https://maven.apache.org/guides/mini/guide-proxies.html ? If it doesn't work, could you try to pass `-D` options?

> From another approach, why the sphinx binaries are not deployed on a maven repository, in a more standard way, so that we could reuse maven settings.xml / default proxy configuration ?

That's a good question. I just didn't do it because it involved an additional manual step, i.e. uploading to Maven central from my local machine, because I do not want to share my PGP key and oss.sonatype.org credentials with CI vendors. I think I can write some scripts that automates the upload process to some extent, though. Let me try when I have some spare cycles.

**Comment by trustin** [comment]
Ah, now I remember why. One reason I did not use Maven repository was because I wanted to download the Sphinx binary based on the current operating system. Adding the dependencies to the binary files for all supported OSes would mean a lot more traffic than necessary.

**Comment by trustin** [comment]
Oh, and please share your proxy settings in `~/.m2/settings.xml` if it didn't work as expected, so I can add more fixes.

**Comment by donatelloOo** [comment]
Please see my comments on the 438d12c commit page.

**Comment by trustin** [comment]
Thanks for your help.

**Comment by trustin** [comment]
Just released 2.2.1
